docs/development/design/quic-driver.md

```diff
@@ -0,0 +1,156 @@
+# QUIC based tunnel driver for Nexodus
+
+> [Issue #1079](https://github.com/nexodus-io/nexodus/issues/1079)
+
+## Summary
+
+FIPS compliance is one of the major limitations of wireguard. This limitation can be a challenge for some users to adopt the wireguard based solution. QUIC protocol can be a viable alternative that can be used to build an alternate option that can support wireguard-like tunnelling and is FIPS compliant as well. QUIC is a new UDP based protocol that is designed to be secure and reliable. It is also designed to be FIPS compliant. This proposal is to create a QUIC based tunnel driver for Nexodus, that is feature compatible with the wireguard driver.
+
+## Requirements
+
+Before diving into the proposed solution, I think it is important to list out the requirements for the QUIC based tunnel driver. These requirements are pretty much derived from wireguard supported features.
+
+- Tunnel creation across
+  - public cloud
+  - private cloud
+  - on-prem (including devices behind home router)
+- Support tunnel creation in above scenarios with
+  - an easy NAT in place (UDP hole punching)
+  - a symmetric NAT in place (through a relay server)
```

Comment:
- Do we add mutli-point from a device's tun interface here? 
- I feel like the other bare bones requirement is being able to source a STUN from the device's interface source port e.g. `SO_REUSEPORT`.

Reply:
Yeah, I was planning to add more details about the feature set (mainly discovery), but i thought thats probably getting more into the architecture level details, so i was thinking of adding those details in the architecture document. But i can add some more details here, hopefully that will help clarify it further.